cmd/tailscale/backend.go

```diff
@@ -313,3 +327,92 @@ func (b *backend) SetupLogs(logDir string, logID logtail.PrivateID) {
 		log.Printf("SetupLogs: filch setup failed: %v", filchErr)
 	}
 }
+
+func (b *backend) logDnsConfigMechanisms() {
+	var baseConfig string
+	err := jni.Do(b.jvm, func(env *jni.Env) error {
+		cls := jni.GetObjectClass(env, b.appCtx)
+		m := jni.GetMethodID(env, cls, "getDnsConfigFromLinkProperties", "()Ljava/lang/String;")
+		n, err := jni.CallObjectMethod(env, b.appCtx, m)
+		baseConfig = jni.GoString(env, jni.String(n))
+		return err
+	})
+	if err != nil {
+		log.Printf("getDnsConfigFromLinkProperties JNI: %v", err)
+	} else {
+		log.Printf("getDnsConfigFromLinkProperties: %s", baseConfig)
+	}
+
+	err = jni.Do(b.jvm, func(env *jni.Env) error {
+		cls := jni.GetObjectClass(env, b.appCtx)
+		m := jni.GetMethodID(env, cls, "getDnsServersFromSystemProperties", "()Ljava/lang/String;")
```

Comment:
factor out these JNI line(s)?

Reply:
Moved the loop inside JNI.Do. It creates a temporary JNI environment, we can't hold Java objects across multiple calls to JNI.Do().